lib/models/package-info-cache/package-info.js

```diff
@@ -415,6 +490,126 @@ class PackageInfo {
+
+    let addon;
+
+    try {
+      addon = new AddonConstructor(parent, project);
+    } catch (e) {
+      if (parent && parent.ui) {
+        parent.ui.writeError(e);
+      }
+      const SilentError = require('silent-error');
+      throw new SilentError(`An error occurred in the constructor for ${this.name} at ${this.realPath}`);
+    }
+
+    AddonConstructor._meta_.initializeIn = Date.now() - start;
+    addon.constructor = AddonConstructor;
+
+    return addon;
+  }
+
+  /**
+   * Create an instance of the addon represented by this packageInfo or (if we
+   * are supporting singletons and this is a singleton-enabled addon) check if
+   * we should be creating a proxy instead.
+   * NOTE: we assume that the value of 'isSingleton' does not change between
+   * calls to the constructor! A given addon is either a singleton or not for an
+   * entire run.
+   *
+   * @method getAddonInstance
+   * @param {} parent the addon/project that is to be the direct parent of the
+   * addon instance created here
+   * @param {*} project the project that is to contain this addon instance
+   * @return {Object} the constructed instance of the addon
+   */
+  getAddonInstance(parent, project) {
+    let ctor = this.getAddonConstructor();
+    let addon;
+
+    if (!isSupportingSingletons || !ctor.isSingleton) {
+      // We have a 'regular' addon or we're not supporting singleton addons at all,
+      // so just return a regular instance. It is not in any singleton cache.
+      addon = this.constructAddonInstance(parent, project);
+
+      if (addon.initializeAddons) {
+        addon.initializeAddons();
+      } else {
+        addon.addons = [];
+      }
+
+      return addon;
+    }
+
+    // Find the bundleRoot that will contain this singleton instance.
+    // Find some parent, if there is one, that returns true for PerBundleAddonCache.isBundleRoot(obj).
+    // 'Project' is always the ultimate bundle.
+
+    // An optimization that lazy engines does that we can do as well. If the bundle
+    // root is not the project, but the project does or is going to have an instance
+    // of this singleton in its own cache, then use the project cache.
+    let bundleRoot = this.findBundleRoot(parent);
+    let bundledProjectAddons = project.getBundledAddonPackageInfos();
+
+    if (bundleRoot === project || (bundleRoot !== project && !bundledProjectAddons.has(this))) {
+      // the instance belongs to its bundle root, whether it's the project or not.
+
+      // Luckily for us, both types of bundle root (Project and lazy engine) have a 
+      // _packageInfo field, just what we need!
+      let bundleRootPkgInfo = bundleRoot._packageInfo;
+
+      let cacheEntry = project.perBundleAddonCache.getPerBundleCacheEntry(bundleRootPkgInfo, this);
+
+      if (cacheEntry && cacheEntry.isBuilt) {
+        // Return a proxy for the already-set-up cache entry.
+        console.log(`BR ${bundleRootPkgInfo.realPath}: PROXY: ${this.realPath}`);
```

Comment:
Note: ✂️

Reply:
Converted these to `logger.debug` statements.